apps/server/src/modules/task/uc/task.uc.spec.ts

```diff
@@ -91,449 +70,401 @@ describe('TaskUC', () => {
-			const spy = lessonRepo.findAllByCourseIds.mockResolvedValue([lessons, lessons.length]);
-			return spy;
-		},
-	};
+	afterEach(() => {
+		jest.resetAllMocks();
+	});
 
 	it('should be defined', () => {
 		expect(service).toBeDefined();
 	});
 
-	const findAllMock = (data?: {
-		tasks?: Task[];
-		lessons?: Lesson[];
-		courses?: Course[];
-		hasWritePermission?: boolean;
-	}) => {
-		const spy1 = setTaskRepoMock.findAllFinishedByParentIds(data?.tasks);
-		const spy2 = setCourseRepoMock.findAllForTeacherOrSubstituteTeacher(data?.courses);
-		const spy3 = setCourseRepoMock.findAllByUserId(data?.courses);
-		const spy4 = setCourseRepoMock.findAllForTeacher(data?.courses);
-		const spy5 = setLessonRepoMock.findAllForTeacher(data?.lessons);
-		const spy6 = setAuthorizationServiceMock.getUserWithPermissions();
-		const spy7 = setUserRepoMock.findById();
-		setTaskRepoMock.findAllByParentIds(data?.tasks);
-
-		const mockRestore = () => {
-			spy1.mockRestore();
-			spy2.mockRestore();
-			spy3.mockRestore();
-			spy4.mockRestore();
-			spy5.mockRestore();
-			spy4.mockRestore();
-			spy6.mockRestore();
-			spy7.mockRestore();
-		};
-
-		return mockRestore;
-	};
 	describe('findAllFinished', () => {
-		beforeEach(() => {
-			const permissions = [Permission.TASK_DASHBOARD_VIEW_V3];
-			user = setupUser(permissions);
-		});
-
-		it('should call auth getUserWithPermissions() with userId', async () => {
-			const mockRestore = findAllMock();
-
-			await service.findAllFinished(user.id);
+		describe('when user has read permission', () => {
+			describe('When task has user parent and is found successfully', () => {
+				const setup = () => {
+					const permissions = [Permission.TASK_DASHBOARD_VIEW_V3];
+					const user = setupUser(permissions);
+					const task = taskFactory.finished(user).build();
 
-			expect(authorizationService.getUserWithPermissions).toHaveBeenCalledWith(user.id);
+					authorizationService.getUserWithPermissions.mockResolvedValueOnce(user);
+					courseRepo.findAllByUserId.mockResolvedValueOnce([[], 0]);
+					lessonRepo.findAllByCourseIds.mockResolvedValueOnce([[], 0]);
+					lessonRepo.findAllByCourseIds.mockResolvedValueOnce([[], 0]);
+					authorizationService.hasPermission.mockReturnValueOnce(false);
+					taskRepo.findAllFinishedByParentIds.mockResolvedValueOnce([[task], 1]);
 
-			mockRestore();
-		});
-
-		it('should return task for a user', async () => {
-			const task = taskFactory.finished(user).build();
-			const mockRestore = findAllMock({ tasks: [task] });
-
-			const [, total] = await service.findAllFinished(user.id);
-
-			expect(total).toEqual(1);
-
-			mockRestore();
-		});
-
-		it('should call task repo findAllFinishedByParentIds', async () => {
-			const mockRestore = findAllMock({});
-			const spy = setTaskRepoMock.findAllFinishedByParentIds();
+					return { user, task };
+				};
 
-			await service.findAllFinished(user.id);
+				it('should call auth getUserWithPermissions() with userId', async () => {
```

Comment:
The relevant call to the auth service is `checkOneOfPermissions` which is missing in the tests. 

I think you could leave out the test for `getUserWithPermissions` because it is a hard requirement for the code below it.

Reply:
Not in focus here. Moved to miro board as todo